example_gh_test.go

```diff
@@ -70,29 +72,66 @@ func ExampleRESTClient_request() {
 	if err != nil {
 		log.Fatal(err)
 	}
-
 	// URL to cli/cli release v2.14.2 checksums.txt
 	assetURL := "repos/cli/cli/releases/assets/71589494"
-	resp, err := client.Request("GET", assetURL, nil)
+	response, err := client.Request(http.MethodGet, assetURL, nil)
 	if err != nil {
 		log.Fatal(err)
 	}
-	defer resp.Body.Close()
-
+	defer response.Body.Close()
 	f, err := os.CreateTemp("", "*_checksums.txt")
 	if err != nil {
 		log.Fatal(err)
 	}
 	defer f.Close()
-
-	_, err = io.Copy(f, resp.Body)
+	_, err = io.Copy(f, response.Body)
 	if err != nil {
 		log.Fatal(err)
 	}
-
 	fmt.Printf("Asset downloaded to %s\n", f.Name())
 }
 
+// Get releases from cli/cli repository using REST API with paginated results.
+func ExampleRESTClient_pagination() {
+	var linkRE = regexp.MustCompile(`<([^>]+)>;\s*rel="([^"]+)"`)
+	findNextPage := func(response *http.Response) (string, bool) {
+		for _, m := range linkRE.FindAllStringSubmatch(response.Header.Get("Link"), -1) {
```

Comment:
Thought for the future: having to do this as an extension author feels a bit low-level and requires understanding of certain details in the GitHub API. I wonder if we could provide facilities to encapsulate pagination from go-gh in the future?

Reply:
Agreed https://github.com/cli/go-gh/issues/23